apps/ex_nvr/lib/ex_nvr/remote_storages/store.ex

```diff
@@ -0,0 +1,18 @@
+defmodule ExNVR.RemoteStorages.Store do
+  @moduledoc false
+
+  alias ExNVR.Model.{Device, Recording, RemoteStorage}
+  alias ExNVR.RemoteStorages.Store.{HTTP, S3}
+
+  @callback save_recording(Device.t(), Recording.t(), opts :: Keyword.t()) ::
+              :ok | {:error, any()}
+
+  @callback save_snapshot(RemoteStorage.t(), binary(), DateTime.t(), binary()) ::
```

Comment:
Try to make it generic as much as possible, pass the config of the handler in the opts keyword. Also give a name to the `binary()`, we don't know what it means
```elixir
@callback save_snapshot(Device.t(), snapshot :: binary(), timestamp :: DateTime.t(), opts :: Keyword.t()) :: :ok | {:error, any()}
```